requirements.txt

```diff
@@ -1,13 +1,26 @@
-# Common Requirements
--r requirements_common.txt
-
-# Find older pytorch pip wheel from PyTorch Org website
---find-links https://download.pytorch.org/whl/torch_stable.html
-
-# Torch CPU Version Windows/Linux
-torch==1.8.0+cpu; sys_platform != "darwin"
-torchvision==0.9.0+cpu; sys_platform != "darwin"
-
-# Torch CPU Version macOS (macOS has no CPU/GPU version separation)
-torch==1.8.0; sys_platform == "darwin"
-torchvision==0.9.0; sys_platform == "darwin"
\ No newline at end of file
+# workarounds for pip 20.0.2's buggy dependency resolver
+# which doesn't account for cross-package constraints: https://pip.pypa.io/en/stable/user_guide/#changes-to-the-pip-dependency-resolver-in-20-3-2020
+# matplotlib pulls in the latest pyparsing, nibabel pulls in the latest packaging, and the two latests are in conflict.
+# > ERROR: packaging 21.2 has requirement pyparsing<3,>=2.0.2, but you'll have pyparsing 3.0.5 which is incompatible.
+# This forces the older pip, if the user has the older pip, to behave itself.
+# This is to specially support users on Ubuntu 20.04 LTS; when Ubuntu 22.04 LTS comes out, this can be removed.
+pyparsing<3,>=2.0.2
+csv-diff>=1.0
+loguru~=0.5
+joblib~=1.0
+matplotlib>=3.3.0
+nibabel~=3.2
+onnxruntime==1.7.0
+pandas~=1.1
+pillow>=7.0.0
+pybids>=0.14.0
+scikit-learn>=0.20.3
+scikit-image~=0.17
+seaborn~=0.11
+tensorboard>=1.15.0
+tqdm>=4.30
+scipy
+torchio>=0.18.68
+torch==1.8.1
+torchvision==0.9.1
```

Comment:
Notice: these two lines were merged from the previous split-brained lines in `requirements_{gpu,cpu}.txt` into the old `requirements_common.txt`, and now everything fits in a single file.